### ISSUE TITLE
[Messenger] Ability to use pconnect on Redis bridge

### ISSUE THREAD

**Issue by xepozz** [symfony/symfony#44804]
### Description

How about to add an ability to use persistence connect to Redis?
I mean to use configurable method in [this line](https://github.com/symfony/symfony/blob/5.3/src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php#L119) `pconnect` or `connect` like it works in cache adapter.
Thanks!

**Comment by bhavin-nakrani** [comment]
I think you can set "persistent" in configuration option. https://symfony.com/doc/current/components/cache/adapters/redis_adapter.html#configure-the-options

**Comment by xepozz** [comment]
I am not about cache adapter

**Comment by xepozz** [comment]
Any updates guys?

**Comment by nicolas-grekas** [comment]
PR welcome @xepozz.

**Comment by yyeltsyn** [comment]
As a workaround I tried to use the existing redis connection:
```
        'transport_factory' => fn (ContainerInterface $container, ParameterBag $config, SerializerInterface $serializer) => function (string $priority) use ($container, $config, $serializer): TransportInterface {
            $queueDsn = $config->get("queues.{$priority}_priority", fn () => $config->mustGet('queues.dsn'));
            $redisDsn = \substr($queueDsn, 0, \strpos($queueDsn, '?'));
            $redisDsn .= "?persistent=1&compression=1&persistent_id={$priority}-priority-queue";

            $connection = Connection::fromDsn(
                $queueDsn,
                [
                    'stream' => "{$priority}_priority_tasks",
                    'group' => $container->get('redis_consumer_group'),
                    'consumer' => $container->get('redis_consumer'),
                ],
                RedisAdapter::createConnection($redisDsn),
            );

            return new RedisTransport($connection, $serializer);
        },
```

But it makes new connection anyway exactly in the line @xepozz mentioned. Maybe it makes sense to fix it to 
```
        if (!$redis->isConnected()) {
            $redis->connect($host, $port);
        }  
```

**Comment by nicolas-grekas** [comment]
This has been implemented in 6.1. You'll need to provide a `persistent_id`.

**Comment by magnetik** [comment]
As a note (I should open an issue) : from a DX point of view it's surprising that the option "persistent" exists in the cache part but no in the messenger part.
